docs-markdown/src/helper/metadata.ts

```diff
@@ -0,0 +1,48 @@
+import { updateMetadataDate } from '../controllers/metadata-controller';
+import { workspace, window } from 'vscode';
+import { noActiveEditorMessage, toShortDate, isMarkdownFileCheck } from './common';
+import { findReplacement } from './utility';
+
+const blacklist: string[] = [];
+
+export async function nag() {
+	if (workspace.getConfiguration('markdown').metadataNag === false) {
+		return;
+	}
+	const editor = window.activeTextEditor;
+	if (!editor) {
```

Comment:
Is this catching the case when doing a large Find/Replace?

Reply:
Sorry I'm not sure I understand your question.   Can you please clarify? The line of code you commented on simply returns if there is no active editor.

This function is triggered from an onsave event so if someone were to do a large find/replace it would only trigger the function once IF auto-saved were enabled, assuming the find/replace is an atomic transaction.  

If auto save is not enabled then find/replace should not matter since this is only triggered when save is called.

Reply:
I was wondering if this section of code was protecting against the nag running on a large Find/Replace when run in VS Code. As a fake example, if you do a find for the letter `a`, and replace it with `b`, are you going to have 100+ nags stacked on each other?

Reply:
I tested this locally with auto-save enabled and vscode only triggers 1 onSave event when running a find/replace on the same string. I did add code below to make sure we only nag once per file in a session.  So this code will also prevent the issue you're describing IF vscode did trigger 100 onsave events during a find/replace.

```
// skip files that have already been saved
	if (blacklist.indexOf(fileName) > -1 === true) {
		return;
	}
	blacklist.push(fileName);
```